packages/@uppy/companion-client/src/RequestClient.js

```diff
@@ -75,9 +80,45 @@ module.exports = class RequestClient {
     return res.json()
   }
 
+  preflight (path) {
+    return new Promise((resolve, reject) => {
+      if (this.preflightDone) {
+        return resolve(this.allowedHeaders.slice())
+      }
+
+      fetch(this._getUrl(path), {
+        method: 'OPTIONS'
```

Comment:
I'm actually getting an error on this because `OPTIONS` is not in the CORS whitelist …
![image](https://user-images.githubusercontent.com/1006268/59565599-fba49d00-9055-11e9-934f-bd9171b4d8d5.png)

Request headers:
![image](https://user-images.githubusercontent.com/1006268/59565601-02331480-9056-11e9-9713-74fc66592a27.png)

Response headers:
![image](https://user-images.githubusercontent.com/1006268/59565607-0f500380-9056-11e9-9a3e-4452b6aa25fc.png)

Reply:
I am unable to reproduce the issue locally with firefox. We've [always allowed all methods for all endpoints](https://github.com/transloadit/uppy/blob/master/packages/%40uppy/companion/src/standalone/index.js#L106-L109)

Are you using the companion standalone server when testing? Also are you able to reproduce this with other providers asides S3? Here's my network log when testing with instagram on firefox

<img width="554" alt="Screenshot 2019-06-18 at 11 39 30 AM" src="https://user-images.githubusercontent.com/8383781/59676279-1ba3a000-91bf-11e9-9b0d-62900f7be734.png">

Reply:
Did renee test it against transloadit? Can you? Maybe we're messing something up over there?

Reply:
Ahh…I tested using the examples/aws-companion example in this repo, which does not use the standalone server, and uses the default configuration of the `cors` module. It doesn't do this thing: https://github.com/expressjs/cors#enabling-cors-pre-flight

Reply:
> Did renee test it against transloadit? Can you? Maybe we're messing something up over there?

Transloadit directly uses the standalone server so I wouldn't think we would be messing anything up there.

> Ahh…I tested using the examples/aws-companion example in this repo, which does not use the standalone server

Ah alright, it makes sense now.

Users who have their own personalized express servers are left to determine what methods they allow on their servers, but maybe we should change this and let the companion pluggable app set the allowed methods for its own endpoints 🤔 

Also, should this still be considered a breaking change if it breaks only for users who don't use the standalone server, and happen not to allow the `OPTIONS` method on their server? 🤔

Reply:
i'll check. if the upload actually works despite the CORS failure we should be good. Users will need to update their integration to get new features from minor versions but that's not breaking.

Reply:
Right, the entire upload fails right now if the preflight request fails, maybe we can change it so the preflight request failure is caught, and we use the default allowedHeaders in that case?

Reply:
> Transloadit directly uses the standalone server so I wouldn't think we would be messing anything up there.

That's true but it does pass through HAProxy which does/can tamper with headers. Not the issue right now it seems, but I thought i'd add that as a friendly reminder here to save us a possible headache later.

Reply:
> That's true but it does pass through HAProxy which does/can tamper with headers

Ah ok, I thought you once mentioned that it does almost nothing (for the case of companion) but forward requests to companion, so I was working with that.

Reply:
> maybe we can change it so the preflight request failure is caught, and we use the default allowedHeaders in that case?

will do 👍

Reply:
@goto-bus-stop I have made an update.

PS: I had to force push because I did a rebase to resolve conflicts